delfin/api/schemas/alert_source.py

```diff
@@ -29,7 +29,12 @@
         'auth_protocol': parameter_types.snmp_auth_protocol,
         'privacy_protocol': parameter_types.snmp_privacy_protocol,
         'privacy_key': {'type': 'string', 'minLength': 1, 'maxLength': 255},
-        'engine_id': {'type': 'string', 'minLength': 1, 'maxLength': 255}
+        'engine_id': {'type': 'string', 'minLength': 1, 'maxLength': 255},
+        'context_name': {'type': 'string'},
+        'retry_num': {'type': 'integer'},
+        'expiration': {'type': 'integer'},
+        'port': parameter_types.tcp_udp_port,
+        'validate_config': {'type': 'boolean'},
```

Comment:
why need validate_config? I see many places use it. IMO, it's should not be exposed in the api and saved in database. if SNMPv2 doesn't support the snmp-get validation, we can use version to decide.

Reply:
It is not related to v2 or v3 version. User can configure same or different set of v2/v3 configuration for get/set and trap operations. 
IMO, if user knows it, he can opt for skipping validation. Anyway, By default, validation is enabled. This way we can avoid unnecessary bottleneck which results in configuration failure. We can discuss

Reply:
Btw adding to db or showing this para can be skipped if needed

Reply:
if it's essential, plz define a config param.